### ISSUE TITLE
Add the ability to open individual .glif files

### ISSUE THREAD

**Issue by eliheuer** [linebender/runebender#249]
This I something that came up in this issue: https://github.com/linebender/runebender/issues/248#issuecomment-834614065

If icons are stored in the `resources` directory in the future, it might be nice to store them as individual `.glif` files instead of a full `ufo`. @ctrlcctrlv has set up MFEK to do this, so I'll look into how it is done there at some point.

**Comment by eliheuer** [comment]
It would also be nice if there could be icons that are specific to themes, so one theme might have a completely different icon set loaded from `.glif` files. Maybe some themes could even have bitmap graphics for icons? For example, 3d rendered png icons or low-color palleted pixel art icons with animations and that kind of thing.

**Comment by eliheuer** [comment]
Just keeping the icons in a UFO would probably be way easier for designers, so I'm happy to do that if this is too difficult. It might be a better choice for a user-friendly application anyway.

**Comment by ctrlcctrlv** [comment]
That's not correct because of the nature of the UFO format. Monoliths are less user friendly in this case I feel.

UFO "files" are actually represented as directories. So, almost every file browser is going to just open it and confuse the user. I don't think that MS Windows `explorer.exe` recognizes directories as possibly having extensions? Definitely most GNU/Linux file browsers don't.

So, users are always confused by browsing around and seeing UFO's. Associating each constitutant UFO subfile with a program is the best way in my opinion to fix such an issue. When MFEKufo is done, I can even add a feature in MFEKglif, if the .glif has a parent UFO, "browse parent font".

Soon, what was once an issue can become an asset. User learns about the parts of the UFO file naturally. Click a glyph, get a glyph editor. Click the fontinfo.plist, get a metadata editor. Click the designspace, time to manage axes. And so on.

Achieving this in a monolithic editor is impractical, but perhaps possible if there was very strict separation of concerns. I can't say, since I'm not trying to make a monolithic font editor.

**Comment by eliheuer** [comment]
Honestly, now that I think about it, keeping the icons in a full UFO might be a good way to go, Designers will find it simple and intuitive compared to working with individual `.glif` files. One UFO can go with one theme, simple and easy.

Maybe we could even put the theme text file in a UFO so that themes can be shared as a single UFO file. I imagine this editor might be used by design students in the future, so the idea of packing your theme into one UFO seems like it would be approachable and a good user experience in an education setting like that.

**Comment by ctrlcctrlv** [comment]
I'm thinking about this issue based on its title, not based on the specific case of how to manage your icons. I also do that in a font, have done since very first time I added icons to MFEKglif.

**Comment by ctrlcctrlv** [comment]
The format isn't at all designed for it, I agree, but also think users are going to want to do it—because I want to do it and make fonts :-)

How important you think it is is entirely up to you. I find it very important as glyphs are always being copied and moved around I find when working on variable fonts, or often I'll run a script on the points of an unattached glyph and then want to just see what it looks like, not really edit it. I see MFEKglif as both a _viewer_ and an _editor_, both jobs being equally important. So if there's something to view I try very hard to show _something_.

**Comment by ctrlcctrlv** [comment]
(The something you get being much uglier than what you'd get if the glyph were attached, but still often extremely useful for developing and debugging fonts.) MFEKglif flashes warnings at you currently if you open an unattached glif, I should probably put one in the title bar too. "MFEKglif (UNATTACHED) — /home/user/Workspace/glif.glif"

**Comment by ctrlcctrlv** [comment]
Oh, also, one final thought. You should know that FontForge has used .glif as an available interchange format since it got its UFO support. "Glif" is one of the export options. So, unattached files exist, in large numbers.
